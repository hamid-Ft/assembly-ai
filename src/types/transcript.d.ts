
export interface Entities {
  entity_type: string;
  text: string;
  start?: number;
  end?: number;
}
export interface Words {
  text: string;
  start: number;
  end: number;
  confidence: number;
  speaker: string;
}
export interface Utterance {
  confidence: number;
  end: number;
  speaker: string;
  start: number;
  text: string;
  words: Words[];
}
export interface SentimentAnalysisResults {
  confidence: number;
  text: string;
  sentiment: 'POSITIVE' | 'NEGATIVE' | 'NEUTRAL';
}

export interface Label {
  relevance: number;
  label: string;
}

export interface Result {
  text: string;
  labels: Label[];
  timestamp: {
    start: number;
    end: number;
  };
}

export interface Summary {
  topic: number;
}

export interface IabCategoriesResult {
  status: 'success' | 'unavailable';
  results: Result[];
  summary: Summary;
} 
export interface StatusProps {
  isLoading: boolean;
  status: string;
}


export interface Transcript {
  id: string;
  status?: 'processing' | 'queued' | 'completed' | 'error';
  text?: string;
  sentiment_analysis_results?: SentimentAnalysisResults[];
  entities?: Entities[];
  utterances?: Utterance[];
  iab_categories_result?: IabCategoriesResult;
}
