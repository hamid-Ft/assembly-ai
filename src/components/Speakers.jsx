/* eslint-disable react/prop-types */
const Speakers = ({ transcript }) => {
  return (
    <div>
      <h3>Speakers: </h3>
      <div>
        {transcript.utterances.map((utterance) => (
          <div key={utterance.start}>
            <span style={{ fontWeight: "bolder", paddingRight: "8px" }}>
              {utterance.speaker}:
            </span>
            <span>{utterance.text}</span>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Speakers;
