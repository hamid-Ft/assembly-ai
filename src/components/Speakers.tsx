import React from 'react';
import type { Transcript } from './AudioRecorder';
const Speakers = ({ transcript }: { transcript: Transcript }) => {
  return (
    <div>
      <h3>Speakers: </h3>
      <div>
        {transcript.utterances?.map((utterance) => (
          <div key={utterance.start}>
            <span style={{ fontWeight: 'bolder', paddingRight: '8px' }}>{utterance.speaker}:</span>
            <span>{utterance.text}</span>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Speakers;
