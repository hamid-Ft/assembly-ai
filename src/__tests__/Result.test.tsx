import { render, screen } from '@testing-library/react';
import Speakers from '../components/Speakers';
import { Transcript, Utterance } from '../types/transcript';
import '@testing-library/jest-dom';

describe('Speakers component', () => {
    test('renders Speakers component', () => {
        // Arrange
        const mockTranscript: Transcript = {
            id: '1',
            utterances: [
                {
                    confidence: 0.9,
                    end: 10,
                    speaker: 'Speaker 1',
                    start: 0,
                    text: 'Test text',
                    words: [],
                },
            ] as Utterance[],
        };

        // Act
        render(<Speakers transcript={mockTranscript} />);

        // Assert
        expect(screen.getByText(/Speakers:/i)).toBeInTheDocument();
        expect(screen.getByText(/Speaker 1: Test text/i)).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        // Arrange
        const mockTranscript: Transcript = {
            id: '1',
            utterances: [
                {
                    confidence: 0.9,
                    end: 10,
                    speaker: 'Speaker 1',
                    start: 0,
                    text: 'Test text',
                    words: [],
                },
            ] as Utterance[],
        };

        // Act
        const { asFragment } = render(<Speakers transcript={mockTranscript} />);

        // Assert
        expect(asFragment()).toMatchSnapshot();
    });
});
