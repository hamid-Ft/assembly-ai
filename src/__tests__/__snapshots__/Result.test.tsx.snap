// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speakers component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Speakers: 
    </h3>
    <div>
      <div>
        <span
          class="font-bolder pr-2"
        >
          Speaker 1: Test text
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
